src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -0,0 +1,881 @@
+      end in
+    let sum_coefficients summands_list_opt =
+      Option.map (List.fold_left sum_next_coefficient (None, Z.zero, Z.zero)) summands_list_opt
+    in
+    match sum_coefficients (get_coeff_vec t texp) with
+    | exception _ -> None 
+    | None -> None
+    | Some (var, var_coeff, offset) ->
+      if Option.is_none var then Some (None, offset)
+      else if Z.(var_coeff = one) then Some (var, offset)
+      else None
+
+
+  let get_coeff t texp = timing_wrap "coeff_vec" (get_coeff t) texp
+
+  let abstract_exists var t = match t.d with 
+    | Some d -> {t with d = Some (EArray.reduce_col d (Environment.dim_of_var t.env var))}
+    | None -> t (* there are no  variables in the current environment *)
+
+  let assign_const t var const = match t.d with 
+    | None -> t
+    | Some t_d -> let d = EArray.copy t_d in d.(var) <- (None, const);  {d = Some d; env = t.env}
+
+  let subtract_const_from_var t var const = 
+    match t.d with 
+    | None -> t
+    | Some t_d -> let d = EArray.copy t_d in
+      let subtract_const_from_var_for_single_equality const index element =
+        let (eq_var_opt, off2) = d.(index) in 
+        if index = var then
+          match eq_var_opt with
+          | None -> d.(index) <- (None, Z.(off2 + const))
+          | Some eq_var -> begin if eq_var <> index then d.(index) <- (None, Z.(off2 + const)) end
+        else 
+          begin match eq_var_opt with 
+            | Some eq_var ->
+              if eq_var = var then d.(index) <- (Some eq_var, Z.(off2 - const))
+            | None -> () 
+          end
+      in
+      EArray.iteri (subtract_const_from_var_for_single_equality const) d; {d = Some d; env = t.env}
+end
+
+
+(** TODO: overflow checking *)
+module ExpressionBounds: (SharedFunctions.ConvBounds with type t = VarManagement.t) =
+struct
+  include VarManagement
+
+  let bound_texpr t texpr = None, None(*Some (Z.of_int (-1000)), Some (Z.of_int (1000)) TODO*)
+
+
+  let bound_texpr d texpr1 =
+    let res = bound_texpr d texpr1 in
+    match res with
+    | Some min, Some max ->  if M.tracing then M.tracel "bounds" "min: %s max: %s" (IntOps.BigIntOps.to_string min) (IntOps.BigIntOps.to_string max); res
+    | _ -> res
+
+  let bound_texpr d texpr1 = timing_wrap "bounds calculation" (bound_texpr d) texpr1
+end
+
+module D =
+struct
+  include Printable.Std
+  include ConvenienceOps (Mpqf)
+  include VarManagement 
+
+  module Bounds = ExpressionBounds
+
+  module Convert = SharedFunctions.Convert (V) (Bounds) (struct let allow_global = true end) (SharedFunctions.Tracked) 
+
+  type var = V.t
+
+  let name () = "lin2vareq"
+
+  let to_yojson _ = failwith "ToDo Implement in future"
+
+  let is_bot t = equal t (bot ())
+  let is_bot_env t = t.d = None
+
+  (*this shows "top" for a specific environment to enable the calculations. It is the identity of all equalities*)
+  let identity env = {d = Some (Array.init (Environment.size env) (fun i -> (Some i, Z.zero))); env = env}
```

Comment:
`identity` as the name is a bit confusing. I think `top_of` is the better name in line with the theory of AI.

Reply:
We renamed it because it was confusing for us, especially in context of top_env, top, bot_env, and bot. But if it is somehow weird how we named it we can change it back again

Reply:
ok renamed it